### ISSUE TITLE
updating public copies of the sdk

### ISSUE THREAD

**Issue by dericed** [amiaopensource/decklinksdk#9]
I've been examining the sdk and the eula is more detail and am wondering if it's possible for there to be a public update of the sdk with this method.

- ContributorA goes to blackmagicdesign.com, clicks on the SDK, submits their name and contact info as required, reviews and agrees to the EULA, and then selects `Register & Download`.

- ContributorA then gets to the download page which contains a temporary download link (this contains a Key-Pair-Id and a long base64 Signature value and an Expiration timestamp). Here ContributorA does not download the SDK but only copies the URL and shares the URL with ContributorB.

- ContributorB accesses the URL to download the SDK. ContributorB has not reviewed nor agreed to the EULA and is thus not agreeing to
> The Licensee must not lease, rent, distribute or sub-license the Software, or use the Software in a time-sharing arrangement, or in any other manner that is not the Permitted Use.

- ContributorB unpacks the SDK and reviews the header license of the SDK files which permit distribution and has a common open source boilerplate in the header.

 I am not well versed in associated law/ethics with such EULA, but in the above scenario ContributorA is agreeing to a EULA that applies to the SDK itself and not a temporary URL of the SDK, while the scenario of ContributorB accessing is in a scenario that doesn't include the EULA but they are subject to respect the license within the SDK which unlike the EULA permits sharing.

???

**Comment by kieranjol** [comment]
I want this to work, but I struggle to see how contributer A isn't distributing the SDK and violating the EULA by sharing the URL. Even though the EULA does not explicitly refer to the URL, the sole purpose of the URL is to distribute the SDK. It would be great if this could work though.

**Comment by privatezero** [comment]
Yeah, I am with Kieran on this one - I REALLY want it to work, but I don't think that in a worst case scenario a lawyer would have any trouble nailing someone by saying that distributing the link was  defacto distributing the software.

Maybe if Contributor A left the link open on their computer while they were at lunch and were able to conclusively prove that they had no knowledge of Contributor B downloading it, they might have a case, but that seems to be pretty thin ice!

**Comment by kieranjol** [comment]
Also the lawyer might find this github issue and it might debunk the last
shred of believability from the 'went to lunch, came back and url was
shared' defense.

**Comment by dericed** [comment]
I think Blackmagic is being very confusing here. On one hand the SDK includes a very open and permissive license. Also on https://www.blackmagicdesign.com/products/ultrastudio/software it states:

> Available free with all UltraStudio models is an advanced developer SDK free of charge for Mac and Windows. Develop your own custom solutions to use in house or even sell! UltraStudio also includes DirectShow filters, a WDM Windows Driver and QuickTime API that lets UltraStudio models integrate seamlessly with an unbelievably wide range of applications and codecs. Develop your solutions with confidence because UltraStudio is a flexible, proven foundation that free you to work without limitations. The UltraStudio SDK provides both low level control of hardware and high level interfaces which allow you to easily perform common tasks. It even lets you develop once and then easily deploy your solution across the entire range of UltraStudio capture and playback solutions from Blackmagic Design.

The EULA contradicts this claim.

**Comment by retokromer** [comment]
Contributor_C uses both Blackmagic’s and AJA’s SDK with the hardware she/he bought. The software that he/she implemented using the SDK is then used also on hardware that has regularly be bought by User_D from Blackmagic or AJA. Contributor_C doesn’t feel guilty.

**Comment by dericed** [comment]
I responded to [this thread](https://forum.blackmagicdesign.com/viewtopic.php?f=12&t=52893) with a new post

> I want to point out that the marketing information on the blackmagic website advertises the capture devices as coming with an SDK that allows users to develop custom solutions to sell. For instance on the ultrastudio software page it says "Available free with all UltraStudio models is an advanced developer SDK free of charge for Mac and Windows. Develop your own custom solutions to use in house or even sell!"
> As a customer of Blackmagic devices, I emailed Blackmagic support to find out how I can obtain a copy of the SDK for free that enables me to develop solutions with these permissions and was redirected back to the SDK download page of the support section which requires agreeing EULA that severely restricts my use of the SDK. The EULA required to access the SDK and the marketing material about the SDK appear to contradict each other.
> I hope that Blackmagic can correct this and offer an SDK under the same terms that it uses to advertise the SDK.

We'll see if the moderator approves.

**Comment by retokromer** [comment]
Any news on this?
